WIP PR to help collect feedback on draft

ryancdickson opened the pull request.

ryancdickson committed:
Create version_2_0.md

Create archive record for Version 2.0

ryancdickson committed:
Create ir_version_3_1.md

ryancdickson committed:
Update incident-report.md

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update incident-report.md

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update policy.md

fix incorrect link.

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update incident-report.md

ryancdickson committed:
Update incident-report.md

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update policy.md

BenWilson-Mozilla commented on policy.md:
```diff
@@ -219,7 +230,7 @@ Reports uploaded to the CCADB MUST be publicly available and contain at least th
     - For revised or corrected versions of audit statements made to address errors, omissions, or formatting to a previously completed audit statement, the CA Owner may enter the date of the initial issuance, provided that the scope and audit period covered by the revised audit statement remain unchanged. In such cases, the audit statement should contain the initial issuance date and the subseque ...
 12. For ETSI, a statement to indicate if the audit was a full audit, and which parts of the criteria were applied, e.g. DVCP, OVCP, NCP, NCP+, LCP, EVCP, EVCP+, QCP-w, Part1 (General Requirements), and/or Part 2 (Requirements for trust service providers), and a statement to indicate that the auditor referenced the applicable CA/Browser Forum criteria and the version used;
-13. All incidents disclosed by the CA Owner, or reported by a third party, and all findings reported by an auditor, that, at any time during the audit period, occurred, were open in Bugzilla, or were reported to a Root Store Operator; and
+13. All [incidents](#61-disclosing-and-responding-to-incidents) disclosed by the CA Owner, or reported by a third party, and all findings reported by an auditor, that, at any time during the audit period, occurred, were open in Bugzilla, or were reported to a Root Store Operator. For each, auditors SHOULD opine that (a) the scope, impact, and root cause of incidents are accurately and fairly stat ...
```
I'd rephrase this to sound more neutral by replacing the word "opine" and also less predictive in terms of remediation ("meaningfully reduce likelihood of recurrence"). Here is some alternative language we could consider.
"For each, auditors SHOULD review the publicly-disclosed incident reports for consistency with audit evidence obtained" ... "and indicate [whether/that] (a) the scope, impact, and root cause are accurately and fairly described;" and "[whether/that] (b) the corrective actions described by the CA Owner are aligned with the factors that led to the incident and are intended to mitigate the risks associated with the identified root cause(s)."

ryancdickson replied:
Incorporated in https://github.com/ryancdickson/www.ccadb.org/commit/119f57da1d44a6c59d24aaa7d61791c970fd3154.

ryancdickson committed:
Update policy.md

ryancdickson committed:
Update policy.md

ChristopherRC committed:
nit: uppercase Auditors

ChristopherRC commented on cas/incident-report.md:
```diff
@@ -79,12 +81,15 @@ For publicly-trusted CA Owners, the number of incident reports filed in Bugzilla
+- other matters
+
+The ETSI Audit scheme commonly, but not exclusively, presents these as:
+- non-conformities (e.g., major, minor, etc.)
```
```suggestion
- non-conformities (sometimes prefaced by "major" or "minor")
```

We're only aware of these two prefaces to date and we've never seen "major" actually used. It might be beneficial to only stick with what's known and remove the "etc.".

ChristopherRC commented on cas/incident-report.md:
```diff
@@ -231,7 +244,7 @@ While Full Incident Reports SHOULD be posted as soon as possible, they MUST be p
 
 Unless a [Closure Report](#closure-report) has been posted and no further questions have been raised, open incident reports MUST be updated:
-- on or before the "Next update" date in the "Whiteboard" field of the bug (note: CA Owners MAY request the "Next update" Whiteboard field be set by a Root Store Operator to align with a specific date related to an open Action Item.);
+- on or before the "Next update" date in the "Whiteboard" field of the bug (note: CA Owners SHOULD request the "Next update" Whiteboard field be set by a Root Store Operator to align with a specific date related to an open Action Item.);
```
```suggestion
- on or before the "Next update" date in the "Whiteboard" field of the bug and CA Owners SHOULD request the "Next update" Whiteboard field be set by a Root Store Operator to align with a specific date related to an open Action Item;
```
Remove "note" to better encourage and/or make the SHOULD language more prominent.

ChristopherRC commented on policy.md:
```diff
@@ -102,6 +103,7 @@ A subordinate CA certificate is a certificate signed by a root CA or another sub
 
 CA Owners MUST disclose:
 - all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request. Disclosure MUST take place within 7 calendar days of issuance and before the subject CA represented in the certificate begins issuing publicly-trusted certificates.
+- subordinate CA ownership using the "Subordinate CA Owner" field. This is the subordinate CA Owner's name as it appears in the provided audit statements. CA Owners MUST leave this field blank if and only if both control of the private key and domain/IP/email validation activities are performed by the organization listed in the audit statement of the parent certificate.
```
meta question: If something like this is already conveyed within the CCADB either through page layout header guidance or a tooltip or similar - do we want to repeat that language in the CCADB Policy? To me, the CCADB Policy is best for describing things that must go into the CCADB that are not already easily explained in the page layout, like the line above needing to disclose new ICAs and along what timeline. Having the language exist in two places creates opportunities for skew.

ryancdickson replied:
So, let's drop:

"This is the subordinate CA Owner's name as it appears in the provided audit statements. CA Owners MUST leave this field blank if and only if both control of the private key and domain/IP/email validation activities are performed by the organization listed in the audit statement of the parent certificate."

?

ChristopherRC replied:
Maybe we should update the tooltip with Andrew's suggest edit and then change this to:

```suggestion
- subordinate CA ownership using the "Subordinate CA Owner" field, as described in it's tooltip.
```

I fear just leaving "subordinate CA ownership using the "Subordinate CA Owner" field." would result in CA Owners adding their own name incorrectly, as we've seen recently through several CRS inclusion detailed reviews.

ChristopherRC commented on policy.md:
```diff
@@ -228,7 +240,7 @@ For hierarchies used to issue TLS certificates:
 - 2.23.140.1.1
 
-For hierarchies used to issue Code Signing certificates:
+For hierarchies used to issue Code Signing and/pr Time Stamping certificates:
```
```suggestion
For hierarchies used to issue Code Signing and/or Time Stamping certificates:
```

ryancdickson committed:
Update policy.md

Co-authored-by: Chris Clements <cclements@google.com>

ryancdickson committed:
Update cas/incident-report.md

Co-authored-by: Chris Clements <cclements@google.com>

ryancdickson committed:
Update cas/incident-report.md

Co-authored-by: Chris Clements <cclements@google.com>

ChristopherRC commented on policy.md:
```diff
@@ -121,11 +123,18 @@ CA Owners with either (1) a certificate included in a Root Store or (2) a CA cer
+    - The in-scope certificate's SHA-256 fingerprint, and
+    - The set of PKI use cases served.
+- effective **TBD**, state adherance to this policy, and any applicable ecosystem policy the corresponding PKI hierarchy(ies) intends to comply with.
+- effective **TBD**, explicitly and clearly describe the relationship between the document and any other applicable policy documents within the hierarchy (e.g., a superior CA's CP/CPS), including the order of precedence in the event of a conflict.
```
```suggestion

Effective **TBD**, CA policy documents MUST:
- identify the in-scope CAs by including a table or bulleted list in Section 1.3.1 that minimally includes the:
    - in-scope certificate's Subject Name,
    - in-scope certificate's SHA-256 fingerprint, and
    - set of PKI use cases served.
- state adherence to this policy, and any applicable ecosystem policy the corresponding PKI hierarchy(ies) intends to comply with.
- explicitly and clearly describe the relationship between the document and any other applicable policy documents within the hierarchy (e.g., a superior CA's CP/CPS), including the order of precedence in the event of a conflict.
```
Once this date passes, these bullets can roll up into the list above.

ryancdickson committed:
Update CRL disclosure reqs

ryancdickson committed:
Update CRL disclosure requirements for CA certificates

Clarify requirements for CA certificate disclosures regarding CRL URLs, specifying conditions for unexpired and revoked certificates.

ryancdickson committed:
Fix formatting of IssuingDistributionPoint extension requirement

ChristopherRC commented on policy.md:
```diff
@@ -324,20 +345,25 @@ The CCADB [Incident Reporting Guidelines (IRGs)](https://www.ccadb.org/cas/incid
+
+If the CA certificate is expired and there are no longer unexpired leaf certificates validating to it, the CA Owner MUST enter the string `expired` into the "All Full CRL URIs for This Hierarchy" field.
+
+If the CA certificate is revoked, no `crlDistributionPoints` disclosures are required.
```
```suggestion
If the CA certificate is revoked, no `crlDistributionPoints` disclosures are required on that certificate record.
```

ryancdickson committed:
Update policy.md

Co-authored-by: Chris Clements <cclements@google.com>

ryancdickson committed:
Update policy.md

Co-authored-by: Chris Clements <cclements@google.com>

ryancdickson committed:
Update policy.md

Co-authored-by: Chris Clements <cclements@google.com>

ryancdickson committed:
Fix typo in subordinate CA ownership disclosure

ryancdickson committed:
Update effective dates in CA policy documents

dustinhollenback-apple commented on policy.md:
```diff
@@ -122,10 +124,19 @@ CA Owners with either (1) a certificate included in a Root Store or (2) a CA cer
+- state adherence to this policy, and any applicable ecosystem policy the corresponding PKI hierarchy(ies) intends to comply with.
+- explicitly and clearly describe the relationship between the document and any other applicable policy documents within the hierarchy (e.g., a superior CA's CP/CPS), including the order of precedence in the event of a conflict.
+
+To promote simplicity and clarity, all CA policy documents SHOULD:
```
If the documents SHOULD be focused on one specific PKI use case, the exact use cases will need to be listed to avoid confusion, including EKUs that are in scope for each use case. In the case of Apple, we use the term "purpose" in relation to use cases, but those purposes could have different combinations of EKUs or trust bits:
TLS (serverAuth OR serverAuth and clientAuth)
S/MIME (emailProtection OR emailProtection and clientAuth)
Client Auth (clientAuth, but not serverAuth, emailProtection, or timeStamping)
Timestamping (timeStamping and not contain the same public key as any other certificate which asserts any other EKU OIDs)

Since other Root Programs may have different names for these use cases, with different EKUs, it seems like the right time to align on standardized terms that we all use that support all of our different goals. Perhaps expanding into more specific use case labels such as:
TLS Server Authentication-only (serverAuth only)
TLS Server Authentication and Client Authentication (serverAuth and clientAuth only)
S/MIME-only (emailProtection only)
S/MIME and Client Authentication (emailProtection and clientAuth only)
Code Signing-only (codeSigning only)
Code Signing and Time Stamping (codeSigning and timeStamping only)
Client Authentication-only (clientAuth only)
BIMI-only (BrandIndicatorforMessageIdentification only)



Should there be different labels when clientAuth is used along with another EKU? 

What other combinations are there that we recommend get split into separate focused CP/CPS documents?

ChristopherRC commented on policy.md:
```diff
@@ -102,6 +103,7 @@ A subordinate CA certificate is a certificate signed by a root CA or another sub
 
 CA Owners MUST disclose:
 - all subordinate CA certificates capable of validating to a certificate included in a Root Store or associated with a CCADB Root Inclusion Request. Disclosure MUST take place within 7 calendar days of issuance and before the subject CA represented in the certificate begins issuing publicly-trusted certificates.
+- subordinate CA ownership using the "Subordinate CA Owner" field, as described in its tooltip.
```
```suggestion
- subordinate CA ownership using the "Subordinate CA Owner" field. This is the subordinate CA Owner's name as it appears in the provided audit statements. CA Owners MUST leave this field blank if and only if both control of the private key and domain/IP control validation activities are performed by the organization listed in the audit statement of the parent certificate.
```
The tooltip is not visible to everyone. I think the risk of potential skew between the policy and the tooltip is relatively low and this is a fairly important accountability requirement. I revert my previous suggestion.

ryancdickson committed:
Update policy.md

Co-authored-by: Chris Clements <cclements@google.com>

ChristopherRC committed:
Update policy.md

Adds a line about timely population of new fields after being added to the CCADB. This was in response to comments from Rob in the past.

ryancdickson committed:
Revise CA policy document requirements

Removed requirements for identifying in-scope CAs in policy documents.

ChristopherRC committed:
update JSON Array of all Full CRL URIs field name

BenWilson-Mozilla commented on policy.md:
```diff
@@ -324,20 +343,25 @@ The CCADB [Incident Reporting Guidelines (IRGs)](https://www.ccadb.org/cas/incid
-For each unexpired and unrevoked CA certificate record disclosed to the CCADB and within 7 days of the corresponding CA issuing its first certificate, CA Owners MUST disclose either:
-- the URL of a full and complete Certificate Revocation List (CRL); or
-- a JSON Array of Partitioned CRL URLs.
+For any unexpired and unrevoked CA certificate disclosed to the CCADB, CA Owners MUST disclose the complete set of distinct HTTP URLs appearing in the `crlDistributionPoints` extension of the unexpired certificates issued by that CA. The disclosed URLs MUST match exactly as they appear in the issued certificates.
```
What about saying, ", CA Owners MUST disclose**, in a JSON array,** the complete set ..."?

ryancdickson replied:
Is https://github.com/ryancdickson/www.ccadb.org/commit/6763710733aa558444cdd0911318dbc49b5ef1de what you had in mind?

ryancdickson committed:
Change CA certificate URL disclosure to JSON array

Address https://github.com/ryancdickson/www.ccadb.org/pull/3#discussion_r2709625102

ryancdickson committed:
Clarify handling of expired CA certificates

Remove requirement to enter 'expired' for unexpired leaf certificates.

ryancdickson committed:
Update effective date in CCADB policy
